examples/detection/yolov2/yolov2/functions/sum_of_squared_error.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+from __future__ import absolute_import
+
+import numpy as np
+from chainer import function
+
+class SumOfSquaredError(function.Function):
+
+    """Sum of squared error (a.k.a. Euclidean loss) function."""
+
+    def forward_cpu(self, inputs):
+        x0, x1 = inputs
+        self.diff = x0 - x1
+        diff = self.diff.ravel() # batch_size x input_sizeの1次元展開
+        return np.array(diff.dot(diff) / 2, dtype=diff.dtype),
+
+    def forward_gpu(self, inputs):
+        x0, x1 = inputs
+        self.diff = x0 - x1
+        diff = self.diff.ravel()
+        return diff.dot(diff) / diff.dtype.type(2),
+
+    # inputs[0]はx, inputs[1]はt
+    # gyは[1]、xがこれだけ変化した時に、yの変化量を求める。
+    # self.diffは (x - t)の配列
```

Comment:
Modified license can be added like this code.
https://github.com/mitmul/chainer-faster-rcnn/blob/master/lib/faster_rcnn/anchor_target_layer.py